### PR TITLE
Adding +/- for symbolic chmod mode

### PR DESCRIPTION
**Pull request by gnikonorov**
Partial resolution to #417; allow for support of `+` and `-` operators for [chmod](https://ruby-doc.org/stdlib-2.2.2/libdoc/fileutils/rdoc/FileUtils.html#method-c-chmod) and fix an issue where `chmod` was discarding old values for groups when performing permission assignment.

Note the following are still unimplemented and will be added shortly: `Xst`

### CODE REVIEW

File: lib/fakefs/file.rb [hunk 1]
```diff
@@ -758,7 +784,64 @@ def self.convert_symbolic_chmod_to_absolute(mode)
...
+
+    # given the current chmod values for a file return the result of adding or removing chmod_perm_num from the
+    # requested groups permissions ( so performing <GROUP>+<PERMS> or <GROUP>-<PERMS>
+    def self.set_perms_for_group(current_groups_to_vals, group_num, assignment_mode, chmod_perm_num)
+      # get the current recorded mode of the group
+      current_group_mode = current_groups_to_vals[group_num]
+
+      # now that we have the current value of the group, add or remove bits accordingly
+      if assignment_mode == '+'
+        return current_group_mode | chmod_perm_num
```
Comment (grosser) [inline lib/fakefs/file.rb]: don't need these returns

**Comment by gnikonorov** [comment]
@grosser - looks like one of the travis build jobs failed: https://travis-ci.org/fakefs/fakefs/jobs/412924233. Is there a way to re-run travis without submitting another pull request?

**Comment by grosser** [comment]
don't think there is ... repo admins can do re-builds from the UI ... so
kicked on off now

**Comment by gnikonorov** [comment]
Thank you @grosser!

**Pull request merged**